src/test/scala/com/rgoulter/DocumentImplSpec.scala

```diff
@@ -0,0 +1,33 @@
+package com.rgoulter
+
+import org.fife.ui.rsyntaxtextarea.modes.PlainTextTokenMaker
+
+import org.scalatest.FlatSpec
+
+import com.rgoulter.typingtutor.DocumentImpl
+import com.rgoulter.typingtutor.Utils
+
+class DocumentImplSpec extends FlatSpec {
+  "DocumentImpl" should "not consider CR an expected-input character" in {
+    /*
+     * Context/Motivation:
+     *  On Windows, text documents might be loaded with CR/LF
+     *  as the system line ending.
+     *  However, the text area considers 'enter' as '\n',
+     *  and it's not intuitive for the user to enter '\r'
+     *  themselves.
+     */
+    // Assemble
+    val inputTextWithCR = "abc\r\ndef"
+    val tokenMaker = new PlainTextTokenMaker()
+    val tokens = Utils.tokenIteratorOf(inputTextWithCR, tokenMaker)
+    val document = new DocumentImpl(inputTextWithCR, tokens)
```

Comment:
As a side note,
it seems a bit awkward that creating a DocumentImpl requires "knowing" about the TokenMaker,
and then explicitly writing tokens.
It makes for a much "larger interface" that needs to be known before using the class.